src/main/java/net/minestom/server/item/meta/CrossbowMeta.java

```diff
@@ -0,0 +1,190 @@
+package net.minestom.server.item.meta;
+
+import net.minestom.server.item.ItemMeta;
+import net.minestom.server.item.ItemMetaBuilder;
+import net.minestom.server.item.ItemStack;
+import net.minestom.server.item.Material;
+import net.minestom.server.registry.Registries;
+import net.minestom.server.utils.NBTUtils;
+import net.minestom.server.utils.validate.Check;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jglrxavpok.hephaistos.nbt.NBTCompound;
+import org.jglrxavpok.hephaistos.nbt.NBTList;
+import org.jglrxavpok.hephaistos.nbt.NBTTypes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class CrossbowMeta extends ItemMeta implements ItemMetaBuilder.Provider<SpawnEggMeta.Builder> {
+
+    private final boolean triple;
+    private final ItemStack projectile1, projectile2, projectile3;
+    private final boolean charged;
+
+    protected CrossbowMeta(@NotNull ItemMetaBuilder metaBuilder,
+                           boolean triple,
+                           ItemStack projectile1, ItemStack projectile2, ItemStack projectile3,
+                           boolean charged) {
+        super(metaBuilder);
+        this.triple = triple;
+        this.projectile1 = projectile1;
+        this.projectile2 = projectile2;
+        this.projectile3 = projectile3;
+        this.charged = charged;
+    }
+
+    /**
+     * Gets if this crossbow is charged with 3 projectiles.
+     *
+     * @return true if this crossbow is charged with 3 projectiles, false otherwise
+     */
+    public boolean isTriple() {
+        return triple;
+    }
+
+    /**
+     * Gets the first projectile.
+     *
+     * @return the first projectile, null if not present
+     */
+    public @Nullable ItemStack getProjectile1() {
+        return projectile1;
+    }
+
+    /**
+     * Gets the second projectile.
+     *
+     * @return the second projectile, null if not present
+     */
+    public @Nullable ItemStack getProjectile2() {
+        return projectile2;
+    }
+
+    /**
+     * Gets the third projectile.
+     *
+     * @return the third projectile, null if not present
+     */
+    public @Nullable ItemStack getProjectile3() {
+        return projectile3;
+    }
+
+    /**
+     * Gets if the crossbow is currently charged.
+     *
+     * @return true if the crossbow is charged, false otherwise
+     */
+    public boolean isCharged() {
+        return charged;
+    }
+
+    public static class Builder extends ItemMetaBuilder {
+
+        private boolean triple;
+        private ItemStack projectile1, projectile2, projectile3;
+        private boolean charged;
+
+        /**
+         * Sets the projectile of this crossbow.
+         *
+         * @param projectile the projectile of the crossbow
+         */
+        public Builder projectile(@NotNull ItemStack projectile) {
```

Comment:
I wonder if this (and the other method) should be nullable as there's no way to "remove" the projectile at the moment.

Reply:
Right, but I find it a bit weird for it to be the only place where an item can be null, what about using AIR instead?